rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1935,11 +1939,12 @@ public BooleanWrapper removeNodeSource(String sourceName, boolean preempt) {
     /**
      * {@inheritDoc}
      */
-    public RMState getState() {
+    public org.ow2.proactive.resourcemanager.common.RMState getState() {
         RMStateNodeUrls rmStateNodeUrls = new RMStateNodeUrls(nodesListToUrlsSet(eligibleNodes),
                                                               listAliveNodeUrls(),
                                                               nodesListToUrlsSet(allNodes.values()));
-        RMState state = new RMState(rmStateNodeUrls, maximumNumberOfNodes);
+        org.ow2.proactive.resourcemanager.common.RMState state = new org.ow2.proactive.resourcemanager.common.RMState(rmStateNodeUrls,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "state". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)